doc/nrf/ug_thingy91_gsg.rst

```diff
@@ -85,356 +85,385 @@ The data is transmitted to nRF Cloud.
+      #. In the menu on the left, click :guilabel:`Add file` in the File section, and select :guilabel:`Browse`.
+         A file explorer window appears.
 
-      .. figure:: /images/programmer_com_ports.png
-         :alt: Programmer - COM ports
+         .. figure:: /images/programmer_add_file1.png
+            :alt: Programmer - Add file
 
-         Programmer - COM ports
+            Programmer - Add file
 
-      If the three COM ports are not visible, press ``Ctrl+R`` in Windows or ``command+R`` in macOS to restart the Programmer application.
-      The button text changes to the SEGGER ID of the selected device, and the :guilabel:`Device Memory Layout` section indicates that the device is connected.
+      #. Navigate to the folder you downloaded and extracted from the `Nordic Semiconductor website`_ in the :ref:`Download firmware <download_firmware_thingy91>` step.
+      #. Find the modem firmware zip file with the name similar to :file:`mfw_nrf9160_*.zip` and the number of the latest version and click :guilabel:`Open`.
 
-#. Update the nRF52840 System on Chip (SoC) application:
+         .. note::
+            Do not extract the modem firmware zip file.
 
-   a. Set the SWD selection switch **SW2** to **nRF52**.
-      See `SWD Select`_ for more information on the switch.
+      #. Select the zip file and click :guilabel:`Open`.
+      #. In the Programmer application, scroll down in the menu on the left to :guilabel:`Device` and click :guilabel:`Write`.
 
-   #. In the menu on the left, click :guilabel:`Add file` in the File section, and select :guilabel:`Browse`.
-      A file explorer window appears.
+         .. figure:: /images/programmer_ext_debug_update_modem.png
+            :alt: Programmer - Update modem
 
-      .. figure:: /images/programmer_add_file1.png
-         :alt: Programmer - Add file
+            Programmer - Update modem
+
+         The :guilabel:`Modem DFU` window appears.
+
+         .. figure:: /images/programmer_modemdfu.png
+            :alt: Programmer - Modem DFU
+
+            Programmer - Modem DFU
+
+      #. In the :guilabel:`Modem DFU` window, click :guilabel:`Write`.
+         When the update is complete, a :guilabel:`Completed successfully` message appears.
+
+         .. note::
+            If you have issues updating modem firmware, click :guilabel:`Erase all` before trying to update the modem again. In this case, the contents of the flash memory are deleted and the applications must be reprogrammed.
+
+..
 
-         Programmer - Add file
+.. rst-class:: numbered-step
 
-   #. Navigate to the folder you downloaded and extracted from the `Nordic Semiconductor website`_ in the :ref:`Download firmware <download_firmware_thingy91>` step.
+.. _update_nrf9160_application:
 
-   #. Open the folder that contains the HEX files for updating over USB.
-      See the :file:`CONTENTS.txt` file for information on which file you need.
+Update the nRF9160 SiP application
+==================================
 
```

Comment:
```suggestion
Complete the following steps to program the application firmware:
```
Just for consistency. May be inside or outside the tab.  We could change for step1 (mentioned inside the tab) and step 2 (mentioned outside the tab) accordingly.